Add an "extract method" command.

esteele opened the pull request.

esteele committed:
Add "extract method" command.

JulianEberius commented:
Hi,

Thanks for your pull request. I have merged it. Since it mostly duplicated the code from PythonRefactorRename I refactored the refactoring code and introduced an abstract base class for the two refactorings that can be used in the future for implementing more Rope operations.

Greetings,
Julian

esteele commented:
On Wednesday, January 18, 2012 at 9:49 AM, Julian Eberius wrote:

> Hi,
> 
> Thanks for your pull request. I have merged it. Since it mostly duplicated the code from PythonRefactorRename I refactored the refactoring code and introduced an abstract base class for the two refactorings that can be used in the future for implementing more Rope operations.
> 
> Greetings,
> Julian
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/JulianEberius/SublimeRope/pull/7#issuecomment-3546693

Great. Even better. I'm just lazy. ;)

esteele commented:
Yeesh, and the irony of not properly refactoring a refactoring plugin just hit me...

On Wednesday, January 18, 2012 at 9:49 AM, Julian Eberius wrote:

> Hi,
> 
> Thanks for your pull request. I have merged it. Since it mostly duplicated the code from PythonRefactorRename I refactored the refactoring code and introduced an abstract base class for the two refactorings that can be used in the future for implementing more Rope operations.
> 
> Greetings,
> Julian
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/JulianEberius/SublimeRope/pull/7#issuecomment-3546693
